pkg/grpc/default_api/api.go

```diff
@@ -1130,3 +1137,81 @@ func (a *api) InvokeBinding(ctx context.Context, in *runtimev1pb.InvokeBindingRe
 		Metadata: daprResp.Metadata,
 	}, nil
 }
+
+func (a *api) GetSecret(ctx context.Context, in *runtimev1pb.GetSecretRequest) (*runtimev1pb.GetSecretResponse, error) {
+
+	if a.secretStores == nil || len(a.secretStores) == 0 {
+		err := status.Error(codes.FailedPrecondition, messages.ErrSecretStoreNotConfigured)
+		log.DefaultLogger.Errorf("GetSecret fail,not configured err:%+v", err)
+		return &runtimev1pb.GetSecretResponse{}, err
+	}
+
+	secretStoreName := in.StoreName
+
+	if a.secretStores[secretStoreName] == nil {
+		err := status.Errorf(codes.InvalidArgument, messages.ErrSecretStoreNotFound, secretStoreName)
+		log.DefaultLogger.Errorf("GetSecret fail,not find err:%+v", err)
+		return &runtimev1pb.GetSecretResponse{}, err
+	}
+
+	req := secretstores.GetSecretRequest{
+		Name:     in.Key,
+		Metadata: in.Metadata,
+	}
+
+	getResponse, err := a.secretStores[secretStoreName].GetSecret(req)
+	if err != nil {
+		err = status.Errorf(codes.Internal, messages.ErrSecretGet, req.Name, secretStoreName, err.Error())
+		log.DefaultLogger.Errorf("GetSecret fail,get secret err:%+v", err)
+		return &runtimev1pb.GetSecretResponse{}, err
+	}
+
+	response := &runtimev1pb.GetSecretResponse{}
+	if getResponse.Data != nil {
+		response.Data = getResponse.Data
+	}
+	return response, nil
+}
+
+func (a *api) GetBulkSecret(ctx context.Context, in *runtimev1pb.GetBulkSecretRequest) (*runtimev1pb.GetBulkSecretResponse, error) {
+	if a.secretStores == nil || len(a.secretStores) == 0 {
+		err := status.Error(codes.FailedPrecondition, messages.ErrSecretStoreNotConfigured)
+		log.DefaultLogger.Errorf("GetBulkSecret fail,not configured err:%+v", err)
+		return &runtimev1pb.GetBulkSecretResponse{}, err
+	}
+
+	secretStoreName := in.StoreName
+
+	if a.secretStores[secretStoreName] == nil {
+		err := status.Errorf(codes.InvalidArgument, messages.ErrSecretStoreNotFound, secretStoreName)
+		log.DefaultLogger.Errorf("GetBulkSecret fail,not find err:%+v", err)
+		return &runtimev1pb.GetBulkSecretResponse{}, err
+	}
+
+	req := secretstores.BulkGetSecretRequest{
+		Metadata: in.Metadata,
+	}
+
+	getResponse, err := a.secretStores[secretStoreName].BulkGetSecret(req)
+	if err != nil {
+		err = status.Errorf(codes.Internal, messages.ErrBulkSecretGet, secretStoreName, err.Error())
+		log.DefaultLogger.Errorf("GetBulkSecret fail,bulk secret err:%+v", err)
+		return &runtimev1pb.GetBulkSecretResponse{}, err
+	}
+
+	filteredSecrets := map[string]map[string]string{}
```

Comment:
how about add it when need?

Reply:
It's no longer used here. It's internal to the dapr component. I'll delete it here